### PR TITLE
UCP/API: Add version check mechanism.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by yosefe** [comment]
@shamisp @bosilca

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1111/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/535/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1112/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/536/ for details (Mellanox internal link).

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -391,6 +392,32 @@ void ucp_config_print(const ucp_config_t *config, FILE *stream,
...
+ * @param [out] minor_version       Filled with library minor version.
+ * @param [out] release_number      Filled with library release number.
+ */
+void ucp_get_version(unsigned *major_version, unsigned *minor_version,
+                     unsigned *release_number);
+
+
+/**
+ * @ingroup UCP_CONTEXT
+ * @brief UCP context initialization with particular API version.
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: If this is an internal function, it has to be defined somewhere else ?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1113/ for details.

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (yosefe) [inline src/ucp/api/ucp.h]: It has to be here because it is used by an inline function.

**Comment by yosefe** [comment]
@miked-mellanox IMHO we don't need it for now because we already have method to get version from the library

**Comment by miked-mellanox** [comment]
get_version_string() is about convenience for end-user - easy way to add print into ULP for version.

**Comment by shamisp** [comment]
+1 for get string.

**Comment by shamisp** [comment]
autobot build

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (shamisp) [inline src/ucp/api/ucp.h]: We have to hide this one in doxygen and explicitly document it as an internal function.

**Comment by yosefe** [comment]
@shamisp how can we hide it in doxygen? not put doxygen comment for it?

**Comment by shamisp** [comment]
look for @cond in the header file.

**Comment by shamisp** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1114/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/537/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
the failure is unrelated.

**Comment by miked-mellanox** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1115/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1127/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1134/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1139/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1145/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1161/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1162/ for details.

**Pull request merged**